diff -urN freetype-2.9.orig/include/freetype/config/ftoption.h freetype-2.9/include/freetype/config/ftoption.h
--- freetype-2.9.orig/include/freetype/config/ftoption.h	2018-05-04 20:05:35.286719860 +0300
+++ freetype-2.9/include/freetype/config/ftoption.h	2018-05-04 20:11:19.925999387 +0300
@@ -818,7 +818,7 @@
   /* switch between the two engines using the `hinting-engine' property of */
   /* the cff driver module.                                                */
   /*                                                                       */
-/* #define CFF_CONFIG_OPTION_OLD_ENGINE */
+#define CFF_CONFIG_OPTION_OLD_ENGINE
 
 
   /*************************************************************************/
