diff -urN freetype-2.5.2.orig/include/freetype/config/ftoption.h freetype-2.5.2/include/freetype/config/ftoption.h
--- freetype-2.5.2.orig/include/freetype/config/ftoption.h	2014-03-03 17:43:56.171899693 +0400
+++ freetype-2.5.2/include/freetype/config/ftoption.h	2014-03-03 17:46:05.380375399 +0400
@@ -591,7 +591,7 @@
   /*   This option requires TT_CONFIG_OPTION_BYTECODE_INTERPRETER to be    */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING
 
 
   /*************************************************************************/
