diff -urN freetype-2.9.orig/include/freetype/config/ftoption.h freetype-2.9/include/freetype/config/ftoption.h
--- freetype-2.9.orig/include/freetype/config/ftoption.h	2018-05-04 20:05:35.286719860 +0300
+++ freetype-2.9/include/freetype/config/ftoption.h	2018-05-04 20:09:42.968806673 +0300
@@ -118,7 +118,7 @@
   /* rendering technology that produces excellent output without LCD       */
   /* filtering.                                                            */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
