diff -urN freetype-2.4.11.orig/include/freetype/config/ftoption.h freetype-2.4.11/include/freetype/config/ftoption.h
--- freetype-2.4.11.orig/include/freetype/config/ftoption.h	2013-01-16 12:05:55.489434576 +0400
+++ freetype-2.4.11/include/freetype/config/ftoption.h	2013-01-16 12:09:34.827988475 +0400
@@ -577,7 +577,7 @@
   /*   This option requires TT_CONFIG_OPTION_BYTECODE_INTERPRETER to be    */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING
 
 
   /*************************************************************************/
